### PR TITLE
[Enhancement] remove lock for QueryDetailQueue

### PR DESCRIPTION
**Pull request by kangkaisen**
Why I'm doing:

What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/QueryDetailQueue.java [hunk 1]
```diff
@@ -36,41 +36,48 @@
...
+    private static volatile long latestMSCnt;
+
+
+    static {
+        SCHEDULED.scheduleAtFixedRate(QueryDetailQueue::removeExpiredQueryDetails, 0, 5, TimeUnit.SECONDS);
+    }
 
-    public static synchronized void addAndRemoveTimeoutQueryDetail(QueryDetail queryDetail) {
-        //set event time here to guarantee order
+    public static void addAndRemoveTimeoutQueryDetail(QueryDetail queryDetail) {
```
[the remark this answers is not in the document]
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/qe/QueryDetailQueue.java]: Fix

File: fe/fe-core/src/main/java/com/starrocks/qe/QueryDetailQueue.java [hunk 2]
```diff
@@ -36,41 +36,48 @@
...
 public class QueryDetailQueue {
-    private static final LinkedList<QueryDetail> TOTAL_QUERIES = new LinkedList<QueryDetail>();
+    private static final ConcurrentLinkedDeque<QueryDetail> TOTAL_QUERIES = new ConcurrentLinkedDeque<>();
+    private static final ScheduledExecutorService SCHEDULED = Executors.newSingleThreadScheduledExecutor();
+
 
     //starrocks-manager pull queries every 1 second
     //metrics calculate query latency every 15 second
     //do not set cacheTime lower than these time
     private static final long CACHE_TIME_NS = 30000000000L;
```
[the remark this answers is not in the document]
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/qe/QueryDetailQueue.java]: add a config

File: fe/fe-core/src/main/java/com/starrocks/qe/QueryDetailQueue.java [hunk 3]
```diff
@@ -80,19 +87,17 @@ public static synchronized List<QueryDetail> getQueryDetailsAfterTime(long event
...
     private static long getCurrentTimeNS() {
         long ms = System.currentTimeMillis();
-        if (ms == latestMS) {
+        if (ms == LATEST_MS.get()) {
             latestMSCnt++;
             return ms * 1000000 + latestMSCnt;
         } else {
-            latestMS = ms;
+            LATEST_MS.set(ms);
             latestMSCnt = 0;
```
[the remark this answers is not in the document]
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/qe/QueryDetailQueue.java]: GPT also found this bug before,  fixed.

**Review by packy92 — DISMISSED** [review]

**Review by packy92 — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 4]
```diff
@@ -2111,6 +2111,14 @@ public class Config extends ConfigBase {
     @ConfField(mutable = true)
     public static boolean enable_collect_query_detail_info = false;
 
+    /**
+     *  StarRocks-manager pull queries every 1 second
+     *  metrics calculate query latency every 15 second
+     *  do not set cacheTime lower than these time
+     */
+    @ConfField(mutable = true)
+    public static long query_detail_cache_time_nanosecond = 30000000000L;
```
[the remark this answers is not in the document]
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:2120]: 30_000_000_000L, no this format in Config class.

[the remark this answers is not in the document]
Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:2120]: It seems that I can compile FE after changing it from `30000000000L` to `30_000_000_000L`.

**Review by ZiheLiu — APPROVED** [review]

**Comment by kangkaisen** [comment]
@Mergifyio backport branch-2.5

**Comment by kangkaisen** [comment]
https://github.com/Mergifyio backport branch-3.0

**Comment by kangkaisen** [comment]
https://github.com/Mergifyio backport branch-3.1

**Comment by kangkaisen** [comment]
https://github.com/Mergifyio backport branch-3.2

**Pull request merged**